spec/services/teacher_training_api/publish_provider_checker_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module TeacherTrainingApi
+  RSpec.describe PublishProviderChecker, type: :service do
+    let(:recruitment_cycle_year) { 2023 }
```

Comment:
Maybe stay in step with current cycle?
```suggestion
    let(:recruitment_cycle_year) { Settings.current_recruitment_cycle_year }
```